File: 283-move-zeroes/move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        # Use mult pointers
        # If only length 1, return
        if len(nums) == 1: return
        # Start with pointer at first position and second position
        first = 0
        second = 1
        # Use while loop with second being less than nums length
        while second < len(nums):
        # If first and second zero, advance only second
            if nums[first] == 0 and nums[second] == 0:
                second += 1
                continue
        # If first zero and second nonzero, swap and advance both
            if nums[first] == 0 and nums[second] != 0 :
                nums[first], nums[second] = [nums[second], nums[first]]
        # If first nonzerp and second zero, advance both
        # If first and second nonzero, advance both
            first += 1
            second += 1 

File: 283-move-zeroes/move-zeroes.ts
/**
 Do not return anything, modify nums in-place instead.
 */
function moveZeroes(nums: number[]): void {
    // Use muliple pointers
    // Start with left at 0 and right at 1
    let left = 0;
    let right = 1;
    // Enter while loop and stop when...right is at the end? 
    while(right < nums.length) {
    // If left is 0 and right is nonzero, swap values and increment left
        if(nums[left] === 0 && nums[right] !== 0) {
            [nums[right], nums[left]] = [nums[left], nums[right]];
            left ++;
        } else if (nums[left] !== 0) {
            left ++;
        }
    // Increment right 
        right ++;
    }
};